H-2554, H-1948, H-2396, H-1388: various bug fixes/visual improvements

benwerner01 opened the pull request:
## 🌟 What is the purpose of this PR?

<!-- Explain, at a high level, what this does and why. -->
<!-- Use the 'What does this change?' section to list more specific implementation details. -->

This PR makes several bug fixes/visual improvements in the HASH app, including:
- adding an empty state on the `/settings/organizations` page for when a user isn't a member of any organizations
- fixes a bug where it was possible to archive a draft entity in the `/actions` page and the entity editor, without incoming/outgoing draft links being archived, resulting in a broken state
- always shows the checkbox for selecting all drafts on the `/actions` page, not only when more than one draft has been selected
- fixes long URLs from overflowing the container of the "Info" left sidebar on the `/@[shortname]` page
- switches the order of the `Actions` and `Notifications` sidebar items (`Actions` is now shown above `Notifications`)

## 🔗 Related links

<!-- Add links to any context it is worth capturing (e.g. Issues, Discussions, Discord) -->
<!-- Mark any links which are not publicly accessible as _(internal)_ -->
<!-- Don't rely on links to explain the PR, especially internal ones: use the sections above -->

- H-2554
- H-1948
- H-2396
- H-1388

## 🔍 What does this change?

<!-- Use a bullet list to explain your changes in more detail, if it would be helpful. -->
<!-- If applicable, link to the specific commit.-->

See description.

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?

<!-- Confirm you have taken the necessary action to record a changeset or publish a change, as appropriate -->
<!-- Tick AT LEAST ONE box and delete the rest. Do not delete this section! see libs/README.md for info on publishing -->

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

<!-- If this adds a user facing feature or modifies how an existing feature is used, it likely needs a docs change. -->
<!-- Tick ONE box and delete the rest. Do not delete this section! -->

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

<!-- If this adds or moves an existing package, modifies `scripts` in a `package.json`, it likely needs a turbo graph change. -->
<!-- Tick ONE box and delete the rest. Do not delete this section! -->

The changes in this PR:

- [x] do not affect the execution graph

## ⚠️ Known issues

<!-- Are there known issues / intentionally omitted functionality? Flag them here to save reviewers doing so -->

## 🐾 Next steps

<!-- Are there are planned/suggested follow ups which are related but won't be done in this PR? -->

## 🛡 What tests cover this?

<!-- What automated tests cover this? Existing ones? New ones? None? -->

Manual testing.

## ❓ How to test this?

<!-- Tell reviewers how they can test the functionality -->

Most of the changes are tested by just interacting with the UI.

For testing the bug fix where it was possible to archive a draft entity with outgoing draft links without warning, it would require creating the required drafts via the GraphQL playground.


## 📹 Demo

<!-- Add a screenshot or video showcasing your work -->

<details>
  <summary>The restored warning for when ignoring a draft entity with outgoing/incoming draft links</summary>
 <img width="595" alt="image" src="https://github.com/hashintel/hash/assets/42802102/fc4ed735-d062-4840-8746-64ff8c018f51">
</details>

<details>
  <summary>The always visible "select all" checkbox</summary>
<img width="655" alt="image" src="https://github.com/hashintel/hash/assets/42802102/c4fd72c1-6c76-41b2-9c95-9ac4ae9c0997">
</details>

<details>
  <summary>The empty state of the organizations table</summary>
<img width="847" alt="image" src="https://github.com/hashintel/hash/assets/42802102/e46790f2-ef66-498a-8194-864cd6c3cde4">


</details>

CiaranMn commented on apps/hash-frontend/src/shared/layout/layout-with-sidebar/sidebar.tsx:
```diff
@@ -126,16 +126,16 @@ export const PageSidebar: FunctionComponent = () => {
         tooltipTitle: "",
```
I think we should now make the path for 'Inbox' go to 'Actions' (since that is what in practice people will be using most, also more consistent with the new ordering)

benwerner01 replied:
Good point, have fixed this in https://github.com/hashintel/hash/pull/4656/commits/d842c2a2be54af3169af4ca30efcb43d79e6031c

CiaranMn reviewed (commented):
Thank you @benwerner01!

A couple of small comments

vilkinsons commented on apps/hash-frontend/src/pages/actions.page/draft-entities/draft-entities-context-bar.tsx:
```diff
@@ -107,71 +107,68 @@ export const DraftEntitiesContextBar: FunctionComponent<{
+            hasPartiallySelectedMatchingDraftEntities
+              ? " (all on this page)"
+              : ""}
+            .
```
```suggestion
```

benwerner01 replied:
Addressed in https://github.com/hashintel/hash/pull/4656/commits/323ede31d0e102018e0dcbea0d0cfaf479362f17

vilkinsons commented on apps/hash-frontend/src/pages/actions.page/draft-entities/draft-entities-context-bar.tsx:
```diff
@@ -107,71 +107,68 @@ export const DraftEntitiesContextBar: FunctionComponent<{
+            textTransform: "uppercase",
+          }}
+        >
+          {selectedDraftEntityIds.length} drafts{" "}
```
```suggestion
          {selectedDraftEntityIds.length} items{" "}
```

vilkinsons commented on apps/hash-frontend/src/pages/actions.page/draft-entities/draft-entities-context-bar.tsx:
```diff
@@ -107,71 +107,68 @@ export const DraftEntitiesContextBar: FunctionComponent<{
+            endIcon={<CheckRegularIcon />}
+          >
+            Select all {matchingDraftEntities.length}{" "}
+            {isDefaultFilterState ? "drafts" : "matching current filters"}
```
```suggestion
            {isDefaultFilterState ? "items" : "matching current filters"}
```

benwerner01 replied:
Addressed in https://github.com/hashintel/hash/pull/4656/commits/323ede31d0e102018e0dcbea0d0cfaf479362f17

vilkinsons commented:
A few suggestions above. Two other points. I don't think:

- this addresses H-1947?
- all of H-2396 is addressed by this?

vilkinsons reviewed (changes requested):
See above.

benwerner01 commented:
> this addresses [H-1947](https://linear.app/hash/issue/H-1947)?

It does not, as I explained [internally here](https://hashintel.slack.com/archives/C022217GAHF/p1719591581382619?thread_ts=1719585163.598769&cid=C022217GAHF)

> all of [H-2396](https://linear.app/hash/issue/H-2396) is addressed by this? (sub-bullet 3; my suggests address sub-bullet 1, and Ciaran's prior suggestion 2)

1 and 2 are addressed, I added a comment asking for clarity on what you're referring to regarding 3 - but from your suggested changes I gather you meant to replace the occurrences of the word "draft" with the word "items", so will do that now.

vilkinsons commented:
> > this addresses [H-1947](https://linear.app/hash/issue/H-1947)?
> 
> It does not, as I explained [internally here](https://hashintel.slack.com/archives/C022217GAHF/p1719591581382619?thread_ts=1719585163.598769&cid=C022217GAHF)
> 
> > all of [H-2396](https://linear.app/hash/issue/H-2396) is addressed by this? (sub-bullet 3; my suggests address sub-bullet 1, and Ciaran's prior suggestion 2)
> 
> 1 and 2 are addressed, I added a comment in the ticket earlier today asking for clarity on what you're referring to regarding 3 - but from your suggested changes I gather you meant to replace the occurrences of the word "draft" with the word "items", so will do that now.

Our Linear integration will mark H-1947 as completed if this PR merges because it is still referenced in the PR title. I'll remove it and the link. Thanks for addressing the others.

benwerner01 commented:
@vilkinsons I've replaced the copy where we're mentioning "drafts" with "items" in https://github.com/hashintel/hash/pull/4656/commits/323ede31d0e102018e0dcbea0d0cfaf479362f17 where I thought it made sense.

For the copy:
> You have no drafts currently awaiting review.

I've changed it to:
> You have no actions currently awaiting review.

in this commit:  https://github.com/hashintel/hash/pull/4656/commits/e317ae79b3778651ced20208d955aec1e0803129

benwerner01 commented:
> Our Linear integration will mark [H-1947](https://linear.app/hash/issue/H-1947) as completed if this PR merges because it is still referenced in the PR title. I'll remove it and the link. Thanks for addressing the others.

Sorry, I didn't intend on linking that PR - must've done it by mistake if it was linked previously.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/hashintel/hash?pullRequest=4656) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/hashintel/hash?pullRequest=4656) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/hashintel/hash?pullRequest=4656) it.</sub>
